src/Shared/XMakeAttributes.cs

```diff
@@ -216,13 +218,15 @@ internal static bool TryMergeRuntimeValues(string runtimeA, string runtimeB, out
                 runtimeB = MSBuildRuntimeValues.any;
             }
 
+            string actualCurrentRuntime = GetCurrentMSBuildRuntime();
+
             // if they're equal, then there's no problem -- just return the equivalent runtime.  
             if (runtimeA.Equals(runtimeB, StringComparison.OrdinalIgnoreCase))
             {
                 if (runtimeA.Equals(MSBuildRuntimeValues.currentRuntime, StringComparison.OrdinalIgnoreCase) ||
                     runtimeA.Equals(MSBuildRuntimeValues.any, StringComparison.OrdinalIgnoreCase))
                 {
-                    mergedRuntime = MSBuildRuntimeValues.clr4;
+                    mergedRuntime = actualCurrentRuntime;
```

Comment:
This seems like it might ignore runtimeA and runtimeB if they're the same and MSBuildRuntimeValue.currentRuntime but not the GetCurrentMSBuildRuntime() if that's different.

Reply:
I don't think I understand this question, either. `MSBuildRuntimeValue.currentRuntime` indicates that the task should run on "whatever the current build is running on", so that's inherently whatever's returned from `GetCurrentMSBuildRuntime()`.